apps/openmw/mwinput/sdlmappings.cpp

```diff
@@ -0,0 +1,218 @@
+        keyMap[SDLK_F9] = MyGUI::KeyCode::F9;
+        keyMap[SDLK_F10] = MyGUI::KeyCode::F10;
+        keyMap[SDLK_NUMLOCKCLEAR] = MyGUI::KeyCode::NumLock;
+        keyMap[SDLK_SCROLLLOCK] = MyGUI::KeyCode::ScrollLock;
+        keyMap[SDLK_KP_7] = MyGUI::KeyCode::Numpad7;
+        keyMap[SDLK_KP_8] = MyGUI::KeyCode::Numpad8;
+        keyMap[SDLK_KP_9] = MyGUI::KeyCode::Numpad9;
+        keyMap[SDLK_KP_MINUS] = MyGUI::KeyCode::Subtract;
+        keyMap[SDLK_KP_4] = MyGUI::KeyCode::Numpad4;
+        keyMap[SDLK_KP_5] = MyGUI::KeyCode::Numpad5;
+        keyMap[SDLK_KP_6] = MyGUI::KeyCode::Numpad6;
+        keyMap[SDLK_KP_PLUS] = MyGUI::KeyCode::Add;
+        keyMap[SDLK_KP_1] = MyGUI::KeyCode::Numpad1;
+        keyMap[SDLK_KP_2] = MyGUI::KeyCode::Numpad2;
+        keyMap[SDLK_KP_3] = MyGUI::KeyCode::Numpad3;
+        keyMap[SDLK_KP_0] = MyGUI::KeyCode::Numpad0;
+        keyMap[SDLK_KP_PERIOD] = MyGUI::KeyCode::Decimal;
+        keyMap[SDLK_F11] = MyGUI::KeyCode::F11;
+        keyMap[SDLK_F12] = MyGUI::KeyCode::F12;
+        keyMap[SDLK_F13] = MyGUI::KeyCode::F13;
+        keyMap[SDLK_F14] = MyGUI::KeyCode::F14;
+        keyMap[SDLK_F15] = MyGUI::KeyCode::F15;
+        keyMap[SDLK_KP_EQUALS] = MyGUI::KeyCode::NumpadEquals;
+        keyMap[SDLK_COLON] = MyGUI::KeyCode::Colon;
+        keyMap[SDLK_KP_ENTER] = MyGUI::KeyCode::NumpadEnter;
+        keyMap[SDLK_KP_DIVIDE] = MyGUI::KeyCode::Divide;
+        keyMap[SDLK_SYSREQ] = MyGUI::KeyCode::SysRq;
+        keyMap[SDLK_RALT] = MyGUI::KeyCode::RightAlt;
+        keyMap[SDLK_HOME] = MyGUI::KeyCode::Home;
+        keyMap[SDLK_UP] = MyGUI::KeyCode::ArrowUp;
+        keyMap[SDLK_PAGEUP] = MyGUI::KeyCode::PageUp;
+        keyMap[SDLK_LEFT] = MyGUI::KeyCode::ArrowLeft;
+        keyMap[SDLK_RIGHT] = MyGUI::KeyCode::ArrowRight;
+        keyMap[SDLK_END] = MyGUI::KeyCode::End;
+        keyMap[SDLK_DOWN] = MyGUI::KeyCode::ArrowDown;
+        keyMap[SDLK_PAGEDOWN] = MyGUI::KeyCode::PageDown;
+        keyMap[SDLK_INSERT] = MyGUI::KeyCode::Insert;
+        keyMap[SDLK_DELETE] = MyGUI::KeyCode::Delete;
+        keyMap[SDLK_APPLICATION] = MyGUI::KeyCode::AppMenu;
+
+//The function of the Ctrl and Meta keys are switched on macOS compared to other platforms.
+//For instance] = Cmd+C versus Ctrl+C to copy from the system clipboard
+#if defined(__APPLE__)
+        keyMap[SDLK_LGUI] = MyGUI::KeyCode::LeftControl;
+        keyMap[SDLK_RGUI] = MyGUI::KeyCode::RightControl;
+        keyMap[SDLK_LCTRL] = MyGUI::KeyCode::LeftWindows;
+        keyMap[SDLK_RCTRL] = MyGUI::KeyCode::RightWindows;
+#else
+        keyMap[SDLK_LGUI] = MyGUI::KeyCode::LeftWindows;
+        keyMap[SDLK_RGUI] = MyGUI::KeyCode::RightWindows;
+        keyMap[SDLK_LCTRL] = MyGUI::KeyCode::LeftControl;
+        keyMap[SDLK_RCTRL] = MyGUI::KeyCode::RightControl;
+#endif
+    }
+
+    MyGUI::KeyCode sdlKeyToMyGUI(SDL_Keycode code)
+    {
+        static std::map<SDL_Keycode, MyGUI::KeyCode> keyMap;
+        if (keyMap.empty())
+            initKeyMap(keyMap);
```

Comment:
Do you think it could be better to replace map by switch-case from the performance point of view? Or at least initialize the map by a single statement.

Reply:
Not sure if it is needed. Input system is not a bottleneck in OpenMW.